src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/VersionConverter.cs

```diff
@@ -1,25 +1,111 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Buffers;
+using System.Buffers.Text;
+using System.Diagnostics;
+
 namespace System.Text.Json.Serialization.Converters
 {
     internal sealed class VersionConverter : JsonConverter<Version>
     {
+#if BUILDING_INBOX_LIBRARY
+        private const int MinimumVersionLength = 3; // 0.0
+
+        private const int MaximumVersionLength = 43; // 2147483647.2147483647.2147483647.2147483647
+
+        private const int MaximumEscapedVersionLength = JsonConstants.MaxExpansionFactorWhileEscaping * MaximumVersionLength;
+#endif
+
         public override Version Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
         {
+            if (reader.TokenType != JsonTokenType.String)
+            {
+                throw ThrowHelper.GetInvalidOperationException_ExpectedString(reader.TokenType);
```

Comment:
@layomia completely unrelated to this change, but isn't the point of `ThrowHelper` methods to avoid emitting `throw` opcodes in the main method body?

Reply:
I believe that's indeed the point cc @ahsonkhan @steveharter. We may need a pass at the reader/writer `ThrowHelper` implementation to throw inline rather than return exceptions.

Reply:
I believe that's indeed the point. We may need a pass at the reader/writer `ThrowHelper` implementation to throw inline rather than return exceptions, assuming there have been no JIT changes that render the optimization redundant.